sorting/Bogo_Sort.cpp

```diff
@@ -1,15 +1,25 @@
-// In computer science, bogosort (also known as permutation sort, stupid sort, slowsort, shotgun sort, random sort,
-//monkey sort, bobosort or shuffle sort) is a highly inefficient sorting algorithm based on the generate and test paradigm. The function
-//successively generates permutations of its input until it finds one that is sorted. It is not useful for sorting, but may be used for
-//educational purposes, to contrast it with more efficient algorithms.
-//  Two versions of this algorithm exist: a deterministic version that enumerates all permutations until it hits a sorted one,
-// and a randomized version that randomly permutes its input. 
-//Randomized version is implemented below.
+/**
+ * @file
+ * @brief Bogo sort algorithm
+ *
+ * Implementation Details -
+ *      In computer science, bogosort (also known as permutation sort, stupid sort, slowsort, 
+ *      shotgun sort, random sort, monkey sort, bobosort or shuffle sort) is a highly inefficient 
+ *      sorting algorithm based on the generate and test paradigm.Two versions of this algorithm 
+ *      exist: a deterministic version that enumerates all permutations until it hits a sorted one,
+ *      and a randomized version that randomly permutes its input.Randomized version is implemented here. 
+ *
+ * Algorithm -      
+ *      
+ *      Shuffle the array untill array is sorted.
+ *
+ */
 #include <iostream>
-#include <vector>
+#include <array>
 #include <cassert>
 //This function checks if array is sorted
-bool is_sorted(const std::vector <int> &arr)
+template<typename T, size_t N>
```

Comment:
This function is already available in STL libraries: [`std::is_sorted`](https://en.cppreference.com/w/cpp/algorithm/is_sorted)